modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+URL.
+|===
+
+[discrete]
+== `k8sPatchResource`
+
+It patches any resource in the cluster, based on provided options. When a client needs to perform the partial update, they can use
+k8sPatch. Alternatively can use k8sUpdate to replace an existing resource entirely. See more link:https://datatracker.ietf.org/doc/html/rfc6902[Data Tracker]. It returns a promise that resolves to the response of the resource patched. In case of failure promise gets rejected with HTTP error response.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |Which are passed as key-value pairs in the map.
+
+|`` |options.model - k8s model
+
+|`` |options.resource - The resource to be patched.
+
+|`` |options.data - Only the data to be patched on existing resource
+with the operation, path, and value.
+
+|`` |options.path - Appends as subpath if provided.
+
+|`` |options.queryParams - The query parameters to be included in the
+URL.
+|===
+
+[discrete]
+== `k8sDeleteResource`
+
+It deletes resources from the cluster, based on the provided model, resource. The garbage collection works based on 'Foreground'|'Background', can be configured with propagationPolicy property in provided model or passed in json. It returns a promise that resolves to the response of kind Status. In case of failure promise gets rejected with HTTP error response.
+
+....
+{ kind: 'DeleteOptions', apiVersion: 'v1', propagationPolicy }
+....
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |which are passed as key-value pair in the map.
+
+|`` |options.model - k8s model
+
+|`` |options.resource - The resource to be deleted.
+
+|`` |options.path - Appends as subpath if provided
+
+|`` |options.queryParams - The query parameters to be included in the
+URL.
+
+|`` |options.requestInit - The fetch init object to use. This can have
+request headers, method, redirect, etc. See more \{@link
+https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
+}
+
+|`` |options.json - Can control garbage collection of resources
+explicitly if provided else will default to model's "propagationPolicy".
+|===
+
+[discrete]
+== `k8sListResource`
+
+Lists the resources as an array in the cluster, based on provided options. It returns a promise that resolves to the response.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |Which are passed as key-value pairs in the map
+
+|`` |options.model - k8s model
+
+|`` |options.queryParams - The query parameters to be included in the
+URL and can pass label selector's as well with key "labelSelector".
+
+|`` |options.requestInit - The fetch init object to use. This can have
+request headers, method, redirect, etc. See more \{@link
+https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
+}
```

Comment:
s/etc/and so forth

Formatting in this cell seems off

Reply:
Similar link formatting issue on my local machine as few comments above, but it looks to be correct on [openshift/console API docs](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/docs/api.md#k8slistresource) ?